src/material-experimental/mdc-snack-bar/snack-bar-container.ts

```diff
@@ -0,0 +1,174 @@
+    implements MatSnackBarContainerInterface, AfterViewChecked, OnDestroy {
+  /** Subject for notifying that the snack bar has exited from view. */
+  readonly _onExit: Subject<any> = new Subject();
+
+  /** Subject for notifying that the snack bar has finished entering the view. */
+  readonly _onEnter: Subject<any> = new Subject();
+
+  /** ARIA role for the snack bar container. */
+  _role: 'alert' | 'status' | null;
+
+  private _mdcAdapter: MDCSnackbarAdapter = {
+    addClass: (className: string) => this._setClass(className, true),
+    removeClass: (className: string) => this._setClass(className, false),
+    announce: () => {},
+    notifyClosed: () => this._onExit.next(),
+    notifyClosing: () => {},
+    notifyOpened: () => this._onEnter.next(),
+    notifyOpening: () => {},
+  };
+
+  private _mdcFoundation = new MDCSnackbarFoundation(this._mdcAdapter);
+
+  /** The portal outlet inside of this container into which the snack bar content will be loaded. */
+  @ViewChild(CdkPortalOutlet, {static: true}) _portalOutlet: CdkPortalOutlet;
+
+  /** Element that acts as the MDC surface container which should contain the label and actions. */
+  @ViewChild('surface', {static: true}) _surface: ElementRef;
+
+  /**
+   * Element that will have the `mdc-snackbar__label` class applied if the attached component
+   * or template does not have it. This ensures that the appropriate structure, typography, and
+   * color is applied to the attached view.
+   */
+  @ViewChild('label', {static: true}) _label: ElementRef;
+
+  constructor(
+      private _elementRef: ElementRef<HTMLElement>,
+      public snackBarConfig: MatSnackBarConfig) {
+    super();
+
+    // Based on the ARIA spec, `alert` and `status` roles have an
+    // implicit `assertive` and `polite` politeness respectively.
+    if (snackBarConfig.politeness === 'assertive' && !snackBarConfig.announcementMessage) {
+      this._role = 'alert';
+    } else if (snackBarConfig.politeness === 'off') {
+      this._role = null;
+    } else {
+      this._role = 'status';
+    }
+
+    // `MatSnackBar` will use the config's timeout to determine when the snack bar should be closed.
+    // Set this to `-1` to mark it as indefinitely open so that MDC does not close itself.
+    this._mdcFoundation.setTimeoutMs(-1);
+  }
+
+  ngAfterViewChecked() {
+    // Check to see if the attached component or template uses the MDC template structure,
+    // specifically the MDC label. If not, the container should apply the MDC label class to this
+    // component's label container, which will apply MDC's label styles to the attached view.
+    const attachedViewHasMdcLabel =
+        !!this._label.nativeElement.querySelector('.mdc-snackbar__label');
+    this._label.nativeElement.classList.toggle('mdc-snackbar__label', !attachedViewHasMdcLabel);
+  }
+
+  /** Makes sure the exit callbacks have been invoked when the element is destroyed. */
+  ngOnDestroy() {
+    this._mdcFoundation.close();
```

Comment:
Should we also destroy the foundation here? AFAIK we create a new container each time a snack bar is opened.

Reply:
Added it to the adapter's notifyClosed method. Here in `ngOnDestroy` is too early for it to be destroyed